AnomalyDetection.py

```diff
@@ -26,98 +29,88 @@ def _ComputeScore(coefs, values):
         residual += deviation * deviation
     return math.sqrt(residual / len(values)), slope
 
-def DetectAnomalies(data, windowSize, levels=1, numTopResults=None, visualize=False):
 
+#TODO ktran: improve the algorithm
+def _partition_anomalies(windows, k):
+    """
+    :param windows: windows, sorted by anomaly score in descending order
+    :param k: number of partitions
+    :return: partition positions
+    """
+    diffs = [windows[iw - 1][1] - windows[iw][1]
+             for iw in range(1, len(windows))]
+    top_jump_positions = sorted(range(len(diffs)), key=lambda i: diffs[i], reverse=True)[0:k]
+    return sorted(top_jump_positions, reverse=True)
+
+
+def detect_anomalies(data, window_size, sensitivity=5, num_anomalies=None, visualize=True):
```

Comment:
I would make visualize=False as the default, but not a big deal

Reply:
I keep `visualize=True` to make minimize noise in demos. For Mobius, you will reimplement in C# so it doesn't really matter.